source/UIAHandler/types.py

```diff
@@ -0,0 +1,66 @@
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2022 NV Access Limited
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+
+from typing import (
+	List,
+)
+from typing_extensions import (
+	Protocol,
+)
+
+from comInterfaces.UIAutomationClient import IUIAutomationTextRange
+
+
+class _IUIAutomationTextRangeT(Protocol):
+	# Based on IUIAutomationTextRange
+	# https://docs.microsoft.com/en-us/windows/win32/api/uiautomationclient/nn-uiautomationclient-iuiautomationtextrange
+	# Currently incomplete.

```

Comment:
Might be worth trying this:
https://mypy.readthedocs.io/en/stable/stubgen.html

Reply:
Unfortunately this did not generate helpful types.

The comInterfaces code in `_944DE083_8FB8_45CF_BCB7_C477ACB2F897_0_1_0.py` does not provide a lot of type information.

```py
class IUIAutomationTextRange(comtypes.gen._00020430_0000_0000_C000_000000000046_0_2_0.IUnknown):
    _case_insensitive_ = True
    _iid_ = GUID('{A543CC6A-F4AE-494B-8239-C814481187A8}')
    _idlflags_ = []
```

Running stubgen produces nothing helpful.
`stubgen -m source.comInterfaces._944DE083_8FB8_45CF_BCB7_C477ACB2F897_0_1_0`

Output:
```py
class IUIAutomationTextRange(comtypes.gen._00020430_0000_0000_C000_000000000046_0_2_0.IUnknown): ...
class IUIAutomationTextRangeArray(comtypes.gen._00020430_0000_0000_C000_000000000046_0_2_0.IUnknown): ...
```